sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -142,8 +145,13 @@ export class GlobalEndpointManager {
 
     let location;
     // If we have preferred locations, try each one in order and use the first available one
-    if (this.preferredLocations && this.preferredLocations.length > 0) {
-      for (const preferredLocation of this.preferredLocations) {
+    if (
+      this.preferredLocations &&
+      this.preferredLocations.length > 0 &&
+      startIndex < this.preferredLocations.length
```

Comment:
@v1k1 I have updated the logic to handle cases where the start index falls outside the preferred location range. In such cases, the logic has been modified to select the first element from the locations list.
Do you see any concerns here?